internal/resources/fabric/service_token/models.go

```diff
@@ -128,22 +132,22 @@ func buildUpdateRequest(d *schema.ResourceData) []fabricv4.ServiceTokenChangeOpe
 		for _, connection := range newServiceTokenConnection.(*schema.Set).List() {
 			newBandwidthLimitMap := connection.(map[string]interface{})
 
-			if bandwidth, ok := newBandwidthLimitMap["bandwidthLimit"]; ok {
-				newBandwidthLimit := bandwidth.([]interface{})
-				if len(newBandwidthLimit) > 0 {
-					newAsideBandwidthLimits := converters.IfArrToIntArr(newBandwidthLimit)
-					newAsideBandwidthLimit = newAsideBandwidthLimits[0]
+			if bandwidth, ok := newBandwidthLimitMap["bandwidth_limit"]; ok {
+				if bandwidthLimitValue, ok := bandwidth.(int); ok {
+					newAsideBandwidthLimit = bandwidthLimitValue
+				} else {
+					log.Printf("[DEBUG] Expected bandwidthLimit to be an integer, but got %T", bandwidth)
```

Comment:
I believe Terraform already does a check for this for you so that you don't have to. Could just be unnecessary noise in the code. Did you run into issues with it?

Reply:
Also, you're not returning an error. You're just doing a print. We should remove this type of "code noise" that isn't adding value to the system itself.